dotcom-rendering/src/model/enhance-ad-placeholders.test.ts

```diff
@@ -73,4 +85,43 @@ describe('Enhancing ad placeholders', () => {
 			}
 		},
 	);
+
+	describe('should not insert an ad placeholder after a non text element', () => {
+		const threeParagraphs = getTestParagraphElements(3);
+
+		const elements = [
+			...threeParagraphs,
+			getImageElement(),
+			...threeParagraphs,
+		];
+
+		const input: Block[] = [
+			{
+				...blockMetaData,
+				elements,
+			},
+		];
+
+		const output = enhanceAdPlaceholders(input);
+		const outputElements = getElementsFromBlocks(output);
+		const outputPlaceholders = outputElements.filter(
+			elementIsAdPlaceholder,
+		);
+
+		expect(outputPlaceholders.length).toEqual(1);
+
+		const indexesOfPlaceholders = outputElements.reduce(
+			(idxs: number[], el: FEElement, idx: number) => {
+				if (elementIsAdPlaceholder(el)) {
+					return [...idxs, idx];
+				} else {
+					return idxs;
+				}
+			},
+			[],
+		);
```

Comment:
Technically, this is a closer to a `flatMap`:

```suggestion
		const indexesOfPlaceholders = outputElements.flatMap(
			(el: FEElement, idx: number) =>
				elementIsAdPlaceholder(el) ? [idx] : []
		);
```